### ISSUE TITLE
Shipping cost work only for Europe

### ISSUE THREAD

**Issue by centoasa** [dim00z/gshoppingflux#55]
I had to add the Italy zone (and added within the Italian country) because the courier uses only the zones to apply shipping costs.
So I tried many times, but the Italy zone is not recognized, so all the other countries have shipping costs, but Italy is always 0

**Comment by centoasa** [comment]
I correct myself: It also works for Italy, but we have in our shop also The courier with free shipping (only for Italy) and this courier is always chosen for shipments in Italy, I think because it has the lowest price.

**Comment by Casper-O** [comment]
The module finds the lowest price - in case you don't want a carrier in your feed (Pickup, service point delivery etc), you can exclude it by selecting in gshoppingflux module. Please not It would also exclude this carrier from all countries.

**Comment by centoasa** [comment]
The fundamental criterion of choice should be the courier chosen for that article. This is because it can be very useful to publish articles with courier  "Free Shipping"

**Comment by Casper-O** [comment]
If I understood correctly, you would like to have an option to select some products exported with "Free Shipping", while others use the lowest cost, even that it could also be shipped for free?

**Comment by centoasa** [comment]
For example on this product I've choose 2 courier: 1st free shipping (only for Italy) and UPS for Europe.
![schermata 2018-03-15 alle 12 36 24](https://user-images.githubusercontent.com/5500935/37461121-accc8132-284d-11e8-9276-90a30f5aaaa4.png)
So flux would be publish this product for Italy=0eur and AT,BE,DK.... UPS=20eur

**Comment by Casper-O** [comment]
We acually simulate "Carts" and get the prices by emulating a checkout to each Zone, for the countries in your store. So it should also check if at product is limit to some carriers, like your image.

But per default, all carriers is available for all products, so you would have to limit all products this way, else all products would be free shipping in Italy.

here is the part of the code that do the stuff:
https://github.com/dim00z/gshoppingflux/blob/master/gshoppingflux/gshoppingflux.php#L2570-L2622

**Comment by centoasa** [comment]
in fact it is so: only a few products have free shipping. But if in the list of couriers in flux setting I do not exclude free shipping, flux puts it on all products. Maybe because as I told you I have an Italy zone for Italy country?

**Comment by centoasa** [comment]
I've just tested adding the free shipping courier directly in product, and also setting product with advanced warehouse management (with free shipping as courier), but if I don't exclude free shipping as courier in flux setting, I've all product with free shipping for Italy.

**Comment by nandospc** [comment]
Quoting what centoasa said.
Also, it should be kind of logical to set shipping prices as the best shipping price available for the single product, instead of doing a link to its zone. Imho, that's because the shipping price per zone is not always the same as for the country or, if ams is activated, not always the same as for the associated warehouse (and so, its associated custom courier by extension). Tl;dr the best shipping price should not be intended generally
What we're trying to say is it should be nice to have a sort of implementation to sunder the best shipping price as default (in general, like at the moment) and as single product->associated warehouse->associated courier (best case scenario if we have borderline situations, imho the optimal case).

**Comment by Casper-O** [comment]
I do not know how it works with Advanced Warehouse Management.

without AWM, On each products, you have **Available Carriers** and **Selected Carriers**.
If you as store manager did not change anything - then all carriers is available for the product, including free shipping. So if you want to have some products, that can only use a specifict carrier, you would have to on all others, to select all other carriers.

Product A with "Free shipping Italy"
- You don't need to change anything, Free shipping is showing in feed and would be available for customers

Product B should not have "Free shipping Italy"
- Select "UPS and DHL" - Best price is showed in feed ."Free shipping not available for customers at checkout

**Comment by nandospc** [comment]
The problem is in the case of "Product B", it shows the free shipping into the feed as well, not its best price.
We made a test with two products with AWM unchecked, just to check what you said, and here is the results, look at the screenshots below.
Case of Product A: https://i.imgur.com/KAQWmfc.png
Case of Product B: https://i.imgur.com/yZoJNiA.png
The feed shows 0$, aka free shipping, for both of them and, by extension, for everything into the feed (since it doesn't consider the shipping prices for products with AWM).

Basically, it doesn't work imho.

**Comment by Casper-O** [comment]
Okay, I had a quick look at the code, and it seems like this should be used

`Carrier::getAvailableCarrierList`
instead of 
`Carrier::getCarriersForOrder`

to keep things more clear, would you mind making a new Issue Report related to this specific issue?
